### PR TITLE
ROS 2 support fo Wibotic Wireless Charger

### PR DESCRIPTION
**Pull request by delihus**
### Description
Added publishing WiboticInfo thanks to libuavcan library.

### Requirements

- [x] Code style guidelines followed
- [x] Documentation updated

### Tests 🧪

- [ ] Robot
- [x] Container
- [ ] Simulation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+- `wibotic_info` [*wibotic_msgs/WiboticInfo*]: Wibotic charger measurements.
+
+#### Parameters
+
+- `~can_iface_name` [*string*, default: **can0**]: CAN BUS interface used for Wibotic receiver.
+- `uavcan_node_id_` [*int*, default: **20**]: Uavcan node ID.
+- `uavcan_node_name_` [*string*, default: **can0**]: Uavcan node name.
+- `update_time_s_` [*string*, default: **can0**]: The period of reading WiboticInfo on a CAN BUS.
+
+## Add can interface
```
Comment (rafal-gorecki) [inline README.md]: Should be CAN if you mean CAN bus.
```suggestion
## Add CAN interface
```

File: wibotic_connector_can/package.xml [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
   <name>wibotic_connector_can</name>
-  <version>0.0.1</version>
-  <description>The wibotic_connector_can package</description>
-  <maintainer email="info@wibotic.com">WiBotic Inc.</maintainer>
-  <license>BSD 3-Clause</license>
-  <url type="website">https://www.wibotic.com</url>
+  <version>0.1.0</version>
+  <description>Integration Panther with Wibotic charger</description>
```
Comment (rafal-gorecki) [inline wibotic_connector_can/package.xml]:
```suggestion
  <description>Integration Husarion UGV with Wibotic charger</description>
```

File: LICENSE
Comment (rafal-gorecki) [inline LICENSE:1]: Can we change this LICENSE

File: .github/workflows/run-unit-tests.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
+---
+name: Run unit tests
+
+on:
+  workflow_dispatch:
```
Comment (rafal-gorecki) [inline .github/workflows/run-unit-tests.yaml:5]: IMO Unit test should be on push

**Review by rafal-gorecki — COMMENTED** [review]
Part of review

File: README.md
Comment (rafal-gorecki) [inline README.md:1]: From main README, I'd expect:
- info what it is
- instruction how to build and launch this package.

Current readme like more like ros package description than main readme. I suggest to create README in every pkgs.

File: wibotic_connector_can/include/wibotic_connector_can/uavcan_types/wibotic/WiBoticInfo.hpp [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * UAVCAN data structure definition for libuavcan.
+ *
+ * Autogenerated, do not edit.
+ *
+ * Source file: /home/husarion/ros2_ws/src/wibotic/20200.WiBoticInfo.uavcan
```
Comment (rafal-gorecki) [inline wibotic_connector_can/include/wibotic_connector_can/uavcan_types/wibotic/WiBoticInfo.hpp]: There is no such file. Shouldnt we can script to generete such file?

File: wibotic_connector_can/include/wibotic_connector_can/wibotic_can_driver.hpp [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+  virtual ~WiboticCanDriverInterface() = default;
+
+  /** *
+   * @param can_iface_name The name of the CAN interface.
+   * @param node_id The ID of the node.
+   * @param node_name The name of the node.
+   *
+   * @exception std::runtime_error Thrown if can interface cannot be found.
+   * */
+  virtual void SetUavCanSettings(
```
Comment (rafal-gorecki) [inline wibotic_connector_can/include/wibotic_connector_can/wibotic_can_driver.hpp]: Can you add a description of the class that we are using the UAV can implementation. I think you could just type it here and get rid of `Uav' from the function names.
When I first saw the UAV, I thought about voltage, knowing that the code was about a charger. xd
SetUavCanSettings could be change to ConfigureCAN - it is one time action we do not need to set it multiple times.

File: wibotic_connector_can/src/wibotic_can_driver_node.cpp [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  wibotic_can_driver_->SetUavCanSettings(can_iface_name_, uavcan_node_id_, uavcan_node_name_);
+  wibotic_can_driver_->CreateUavCanNode();
+  wibotic_can_driver_->CreateWiboticInfoSubscriber();
+  wibotic_can_driver_->Activate();
+}
+
+wibotic::WiBoticInfo WiboticCanDriverNode::GetWiboticInfo()
+{
+  const auto update_time_ms = static_cast<std::size_t>(update_time_s_ * 1000);
+  wibotic_can_driver_->Spin(update_time_ms);
```
Comment (rafal-gorecki) [inline wibotic_connector_can/src/wibotic_can_driver_node.cpp:71]: If I understand it right, spin can take up to 1 sec, before it receives an answer. IMO it should spin async with about 10 Hz and return latest measurment, if there was no error in any measurement

**Review by rafal-gorecki — CHANGES_REQUESTED** [review]
Some suggestions

File: LICENSE
Comment (delihus) [reply LICENSE:1]: The library was completely changed. There are no common files. I used fork to keep life of this package.

File: .github/workflows/run-unit-tests.yaml (hunk 3 above)
Comment (delihus) [reply .github/workflows/run-unit-tests.yaml:5]: I just copied from panther_ros. I will add this

File: wibotic_connector_can/include/wibotic_connector_can/wibotic_can_driver.hpp (hunk 5 above)
Comment (delihus) [reply wibotic_connector_can/include/wibotic_connector_can/wibotic_can_driver.hpp]: I would change it to `ConfigureUavCan`. `ConfigureCan`  wouble be more like changing a baudrate of can.

**Review by rafal-gorecki — APPROVED** [review]

**Pull request merged**